code/src/main/java/com/onlinecontacttracing/servlets/get-data/CalendarDataForPositiveUser.java

```diff
@@ -59,18 +85,40 @@ public void run() {
       
       // Iterate through events to extract contacts and places
       for (Event event : events.getItems()) {
-        List<EventAttendee> attendees = Optional.ofNullable(event.getAttendees()).orElse(Collections.emptyList());
-
-        for (EventAttendee attendee : attendees) {
-          contacts.add(new PotentialContact(attendee.getDisplayName(), attendee.getEmail()));
-        }
+        getContactsFromEvent(event);
+        getPlacesFromEvent(event, context);
       }
 
-      // TODO: add positiveUserLocations
+      // Store data or replace old data with newer data.
+      ofy.save().entity(positiveUserPlaces).now();
       
     } catch (Exception e) {
       e.printStackTrace();
       log.warning("An exception occurred: " + e.toString());
     }
   }
+
+  private void getContactsFromEvent(Event event) {
```

Comment:
okay

Reply:
What should the other method return?

Reply:
I decided to pass in an object to the method but that feels a lot like functional programming and it seems that you are hinting at making the more OOP so should I move the method getPlacesFromEvent into the PositiveUserPlaces class?

Reply:
That sounds reasonable. What I really want is for our code to not have "side effects". Calling a method that populates a field quietly isn't obvious. Instead, you want to return that data and explicitly store it, or otherwise make that a top-level feature of how the code is working.